src/main.cpp

```diff
@@ -2497,14 +2497,20 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
 }
 
 
-CConsensusContextInfo GetContextInfo(const CBlockIndex* pindex)
+CConsensusContextInfo GetContextInfo(const CBlockIndex* pindex, const Consensus::Params consensusParams)
 {
     assert(pindex);
     CConsensusContextInfo contextInfo;
     contextInfo.height = pindex->nHeight;
     contextInfo.medianTimePast = pindex->GetMedianTimePast();
     contextInfo.now = GetAdjustedTime();
     contextInfo.bestBlock = pindex->GetBlockHeader();
```

Comment:
I noticed that the consensus code only depends on the current majorityVersion. So I'm calculating the majorityVersion outside consensus, and I make the check inside. I've not found a better solution for now.